### PR TITLE
DDPB-3991 add deputy expected costs

### PR DESCRIPTION
**Pull request by jamesrwarren**
## Purpose
Expected costs behat tests

Fixes DDPB-3991

## Approach
Nothing too crazy here. Just trying out Alex's new expected results stuff. Looked really nice. Couple of tweaks i had to make to make it work with this scenario but otherwise lovely :).

## Learning
Not really. Just learning how to use the new code.

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [x] I have added tests to prove my work
- [ ] The product team have approved these changes

### Frontend
- [ ] I have run an in-browser accessibility test (e.g. WAVE, Lighthouse)
- [ ] There are no deprecated CSS classes noted in the profiler
- [ ] Translations are used and the profiler doesn't identify any missing
- [ ] Any links or buttons added are screen reader friendly and contextually complete
- [ ] If adding GA events, I have updated or [checked](docs/runbooks/GOOGLE-ANALYTICS.md) the existing category or label values

### CODE REVIEW

File: api/tests/Behat/bootstrap/v2/Common/FormFillingTrait.php [hunk 1]
```diff
@@ -282,21 +294,31 @@ public function editAnswerInSectionTrackTotal(NodeElement $summaryRowToEdit, str
...
      *
-     * @throws ElementNotFoundException
+     * @throws BehatException
      */
-    public function editAnswerInSection(NodeElement $summaryRowToEdit, string $fieldName, string $newValue, string $formSectionName): array
+    public function editAnswerInSection(NodeElement $summaryRowToEdit, string $fieldName, $newValue, string $formSectionName, bool $fullGroup = true, string $type = 'field', string $translatedValue = null)
     {
-        $this->removeAnswerFromSection($fieldName, $formSectionName);
-
-        $summaryRowToEdit->clickLink('Edit');
```
Comment (jamesrwarren) [inline api/tests/Behat/bootstrap/v2/Common/FormFillingTrait.php:292]: @acsauk Sorry about this. Just could not get the selection to work properly. Was pulling out my hair so created a helper and changed this to click. Hopefully that's ok or if not maybe you can show me where i was going wrong. Spent over an hour mucking around with that one bit of syntax :( 
This still works with your existing code.

Comment (acsauk) [reply api/tests/Behat/bootstrap/v2/Common/FormFillingTrait.php:292]: I've just had to deal with this in my PR too. While it's nice to have a single function I think having to work out what 7 arguments relate too and use around the codebase might be a step too far. I broke this out to a separate function to be explicit about using it for a select - https://github.com/ministryofjustice/opg-digideps/pull/660/files#diff-62d6fee126d7c272fc2598fcbe612644e274ea3149cea9b44cec999ec9f70285. What do you think?

**Review by acsauk — COMMENTED** [review]
Looks great - one point on the editing a response function but good to merge after that 👍

File: api/tests/Behat/bootstrap/v2/Common/FormFillingTrait.php (hunk 1 above)
Comment (jamesrwarren) [reply api/tests/Behat/bootstrap/v2/Common/FormFillingTrait.php:292]: Yeah can't argue with that. I have a penchant for sticking things in one function that ought to be in two. It's a bad habit!

**Review by Gugandeep — APPROVED** [review]

**Pull request merged**